agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/connectionstring/ConnectionString.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.azure.monitor.opentelemetry.exporter.implementation.connectionstring;
+
+import com.azure.core.util.CoreUtils;
+import com.google.auto.value.AutoValue;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+import javax.annotation.Nullable;
+
+@AutoValue
+public abstract class ConnectionString {
+
+  // visible for testing
+  static final int CONNECTION_STRING_MAX_LENGTH = 4096;
+
+  public static ConnectionString parse(String connectionString) {
+    return parse(connectionString, null, null);
+  }
+
+  public static ConnectionString parse(
+      String connectionString,
+      @Nullable String statsbeatInstrumentationKey,
```

Comment:
I extracted statsbeat out of connection string, but we do still need to update statsbeat here in case the user's connection string has changed from non-EU to EU region

Reply:
that's a good point 👍